### PR TITLE
Customized session in each request

### LINKED ISSUE #100: Customized session in each request
There are cases where you might have to use temporary credentials in order to perform an operation. For instance, when you need to authenticate a user for certain request.

Right now there's already a way to override the current header by doing:
```python
import cuenca

# Configure user credentials
cuenca.session.configure(
    api_key=USER_API_KEY,
    api_secret=USER_SECRET_KEY
)

# Perform operation
cuenca.Card.all(user_id='USlskdjflskdjf')

# Return to default credentials
cuenca.session.configure(
    api_key =CUENCA_API_KEY,
    api_secret=CUENCA_SECRET_KEY
)
```

But it would be clearer and safer doing something like:
```python
import cuenca

with cuenca.configure(
        api_key=USER_API_KEY,
        api_secret=USER_SECRET_KEY
    ) as cuenca_user:
    cuenca_user.Card.all(user_id='USlskdjflskdjf')
```

This is specially useful in `core` which has to make requests on behalf of users to authenticate with their credentials.

### PR DESCRIPTION
**Pull request by ricardo8990**
Agrega soporte para poder utilizar una sesión independiente de las credenciales por defecto

closes #100

### CODE REVIEW

File: cuenca/resources/base.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 )
 
 from ..exc import MultipleResultsFound, NoResultFound
-from ..http import session
+from ..http import Session, session as default_session
```
Comment (matin) [inline cuenca/resources/base.py:14]: rename `default_session` to `global_session`

File: cuenca/resources/base.py [hunk 2]
```diff
@@ -46,19 +46,21 @@ def to_dict(self):
...
     @classmethod
-    def retrieve(cls, id: str) -> Resource:
+    def retrieve(cls, id: str, session: Optional[Session] = None) -> Resource:
+        session = session or default_session
         resp = session.get(f'/{cls._resource}/{id}')
         return cls._from_dict(resp)
 
-    def refresh(self):
-        new = self.retrieve(self.id)
+    def refresh(self, session: Optional[Session] = None):
```
Comment (matin) [inline cuenca/resources/base.py:54]: try this instead:
```python
def refresh(self, session: Session = global_session):
```

File: cuenca/resources/cards.py [hunk 3]
```diff
@@ -25,7 +25,12 @@ class Card(Retrievable, Queryable, Creatable, Updateable):
     status: CardStatus
 
     @classmethod
-    def create(cls, ledger_account_id: str, user_id: str) -> 'Card':
+    def create(
+        cls,
+        ledger_account_id: str,
+        user_id: str,
+        session: Session = global_session,
```
Comment (matin) [inline cuenca/resources/cards.py:32]: Add `*` before this as well

**Review by matin — CHANGES_REQUESTED** [review]
All methods that accept `session` as a kwarg should never accept it as a positional arg. There are still a couple cases that need to be changed. Otherwise, looks good!

File: cuenca/resources/cards.py (hunk 3 above)
Comment (ricardo8990) [reply cuenca/resources/cards.py:33]: Done

**Review by matin — APPROVED** [review]

**Comment by matin** [comment]
@ricardo8990 can you create the release?

**Pull request merged**